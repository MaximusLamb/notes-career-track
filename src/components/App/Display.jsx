import React from 'react';
import propTypes from 'prop-types';

const Display = ({ username, followers, followings, profile }) => (
  <li>
    {username}
    <br></br>
    {followers}
    <br></br>
    {followings}
    <br></br>
    {profile}
  </li>
);

Display.propTypes = {
  username: propTypes.string.isRequired,
  followers: propTypes.number.isRequired,
  followings: propTypes.number.isRequired,
  profile: propTypes.string.isRequired
};

export default Display;
