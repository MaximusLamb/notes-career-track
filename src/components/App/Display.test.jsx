import React from 'react';
import { shallow } from 'enzyme';
import Display from './Display';

describe('Display component', () => {
  it('renders User Display', () => {
    const wrapper = shallow(<Display username="crazyTony" followers={300} following={200} html_url={'http://website.gov'} repos={[{ name: 'crazyTony', html_url: 'http://website.gov' }]}/>);
    expect(wrapper).toMatchSnapshot();
  });
});
  
