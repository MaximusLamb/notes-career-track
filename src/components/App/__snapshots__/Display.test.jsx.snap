// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Display component renders User Display 1`] = `
<ul>
  <br />
  <li>
    FOLLOWERS: 
    300
  </li>
  <br />
  <li>
    FOLLOWING: 
    200
  </li>
  <br />
  <li>
    LINK: 
    <a
      href="http://website.gov"
    >
      http://website.gov
    </a>
  </li>
  <br />
  <li>
    REPOS 
    <ul>
      <li>
        <a />
      </li>
    </ul>
  </li>
</ul>
`;
